crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -0,0 +1,356 @@
+                            view_dimension: wgpu::TextureViewDimension::D2,
+                            multisampled: false,
+                        },
+                        count: None,
+                    },
+                ],
+            },
+        );
+
+        let pipeline_layout = pools.pipeline_layouts.get_or_create(
+            device,
+            &PipelineLayoutDesc {
+                label: "depth_cloud_rp_layout".into(),
+                entries: vec![shared_data.global_bindings.layout, bind_group_layout],
+            },
+            &pools.bind_group_layouts,
+        );
+
+        let shader_module = pools.shader_modules.get_or_create(
+            device,
+            resolver,
+            &ShaderModuleDesc {
+                label: "depth_cloud".into(),
+                source: include_file!("../../shader/depth_cloud.wgsl"),
+            },
+        );
+
+        let render_pipeline = pools.render_pipelines.get_or_create(
+            device,
+            &RenderPipelineDesc {
+                label: "depth_cloud_rp".into(),
+                pipeline_layout,
+                vertex_entrypoint: "vs_main".into(),
+                vertex_handle: shader_module,
+                fragment_entrypoint: "fs_main".into(),
+                fragment_handle: shader_module,
+                vertex_buffers: smallvec![],
+                render_targets: smallvec![Some(ViewBuilder::MAIN_TARGET_COLOR_FORMAT.into())],
+                primitive: wgpu::PrimitiveState {
+                    topology: wgpu::PrimitiveTopology::TriangleList,
+                    ..Default::default()
+                },
+                depth_stencil: ViewBuilder::MAIN_TARGET_DEFAULT_DEPTH_STATE,
+                multisample: wgpu::MultisampleState {
+                    // We discard pixels to do the round cutout, therefore we need to
+                    // calculate our own sampling mask.
+                    alpha_to_coverage_enabled: true,
+                    ..ViewBuilder::MAIN_TARGET_DEFAULT_MSAA_STATE
+                },
+            },
+            &pools.pipeline_layouts,
+            &pools.shader_modules,
+        );
+
+        DepthCloudRenderer {
+            render_pipeline,
+            bind_group_layout,
+        }
+    }
+
+    fn draw<'a>(
+        &self,
+        pools: &'a WgpuResourcePools,
+        pass: &mut wgpu::RenderPass<'a>,
+        draw_data: &'a Self::RendererDrawData,
+    ) -> anyhow::Result<()> {
+        crate::profile_function!();
+        if draw_data.bind_groups.is_empty() {
+            return Ok(());
+        }
+
+        let pipeline = pools.render_pipelines.get_resource(self.render_pipeline)?;
+        pass.set_pipeline(pipeline);
+
+        for (num_points, bind_group) in &draw_data.bind_groups {
+            pass.set_bind_group(1, bind_group, &[]);
+            pass.draw(0..*num_points * 6, 0..1);
+        }
+
+        Ok(())
+    }
+
+    fn draw_order() -> u32 {
+        DrawOrder::Transparent as u32
+    }
```

Comment:
Shouldn't need this anymore.

Reply:
it's gone on `main` anyway - this needs a merge with `main`